tests/Unit/Schema/Directives/UploadDirectiveTest.php

```diff
@@ -0,0 +1,300 @@
+
+        $this->multipartGraphQL(
+            [
+                'query' => /** @lang GraphQL */ '
+                mutation ($file: Upload!) {
+                    file: upload(file: $file)
+                }
+            ',
+                'variables' => [
+                    'file' => null,
+                ],
+            ],
+            ['0' => ['variables.file']],
+            ['0' => $file]
+        )->assertJson([
+            'data' => [
+                'file' => $filePath,
+            ],
+        ]);
+
+        Storage::disk('uploadDisk')->assertExists("/test/{$file->hashName()}");
+    }
+
+    public function testUploadArgumentWhereValueIsNull(): void
+    {
+        $filePath = null;
+
+        $this->mockResolver(static function ($root, array $args) use (&$filePath): ?string {
+            $filePath = $args['file'];
+            return $args['file'];
+        });
+
+        $this->schema = /** @lang GraphQL */
+            '
+        scalar Upload @scalar(class: "Nuwave\\\\Lighthouse\\\\Schema\\\\Types\\\\Scalars\\\\Upload")
+    
+        type Mutation {
+            upload(
+            file: Upload @upload
+            ): String @mock
+        }
+        ' . self::PLACEHOLDER_QUERY;
+
+        $this->multipartGraphQL(
+            [
+                'query' => /** @lang GraphQL */ '
+                mutation ($file: Upload) {
+                    file: upload(file: $file)
+                }
+            ',
+                'variables' => [
+                    'file' => null,
+                ],
+            ],
+            ['0' => ['variables.file']],
+            ['0' => []]
+        )->assertJson([
+            'data' => [
+                'file' => $filePath,
+            ],
+        ]);
+    }
+
+    public function testThrowsAnExceptionIfDiskIsMissing(): void
+    {
+        $this->expectException(Exception::class);
+
+        config()->offsetSet('filesystems.default', 'uploadDisk');
+
+        $this->schema = /** @lang GraphQL */
+            '
+        scalar Upload @scalar(class: "Nuwave\\\\Lighthouse\\\\Schema\\\\Types\\\\Scalars\\\\Upload")
+    
+        type Mutation {
+            upload(
+            file: Upload! @upload(disk:"uploadDisk")
+            ): Boolean
+        }
+        ' . self::PLACEHOLDER_QUERY;
+
+        $file = UploadedFile::fake()->create('test.pdf', 500);
+
+        $this->multipartGraphQL(
+            [
+                'query' => /** @lang GraphQL */ '
+                mutation ($file: Upload!) {
+                    file: upload(file: $file)
+                }
+            ',
+                'variables' => [
+                    'file' => null,
+                ],
+            ],
+            ['0' => ['variables.file']],
+            ['0' => $file]
+        );
+    }
+
+    public function testThrowsAnExceptionIfStoringFails(): void
+    {
+        $this->expectExceptionObject(new CannotWriteFileException("Unable to upload `upload` file to `/` via disk `uploadDisk`"));
```

Comment:
When exactly is this thrown? Please move this expectation as closely as possible to where the exception is triggered.